locust/test/test_main.py

```diff
@@ -1766,34 +2128,58 @@ def t(self):
                     "-",
                     "--worker",
                 ],
-                stderr=STDOUT,
+                stderr=PIPE,
                 stdout=PIPE,
                 text=True,
+                bufsize=1,
             )
-            gevent.sleep(2)
             proc_worker2 = subprocess.Popen(
                 [
                     "locust",
                     "-f",
                     "-",
                     "--worker",
                 ],
-                stderr=STDOUT,
+                stderr=PIPE,
                 stdout=PIPE,
                 text=True,
+                bufsize=1,
             )
-            stdout = proc.communicate()[0]
-            stdout_worker = proc_worker.communicate()[0]
-            stdout_worker2 = proc_worker2.communicate()[0]
 
-            self.assertIn('All users spawned: {"User1": 1} (1 total users)', stdout)
-            self.assertIn("Shutting down (exit code 0)", stdout)
-            self.assertNotIn("Traceback", stdout)
-            self.assertNotIn("Traceback", stdout_worker)
-            self.assertNotIn("Traceback", stdout_worker2)
+            output = []
+
+            poll_until(lambda: all_users_spawned(proc, output), timeout=30)
+
+            stdout, stderr = proc.communicate(timeout=30)
+            stdout_worker, stderr_worker = proc_worker.communicate(timeout=10)
+            stdout_worker2, stderr_worker2 = proc_worker2.communicate(timeout=10)
+
+            output.extend(stdout.splitlines())
```

Comment:
I don't think joining the output of all processes in a single string makes sense, it would be unnecessarily difficult to debug. Can you keep them separate?